File: Python/Warmup-2/no_triples.py

# this function return true if the given array does not contain any triples
# triple = a value appearing 3 times in a row
def no_triples(nums):
	b = True

	for x in range(0,len(nums) - 2):
		if nums[x] == nums[x + 1] and nums[x + 1] == nums[x + 2]:
			b = False

	return b

print(no_triples([1, 1, 2, 2, 1]))
print(no_triples([1, 1, 2, 2, 2, 1]))
print(no_triples([1, 1, 1, 2, 2, 2, 1]))

File: JavaScript/Warmup-2/noTriples.js

// given an array of ints, we'll say that a triple is a value appearing 3 times in a row in the array
// return true if the array does not contain any triples
const noTriples = (nums) => {

    let result = true;
    let found = false;

    if (nums.length > 2) {
        for (let i = 0; i < nums.length - 2 && !found; i++) {
            if (nums[i] === nums[i + 1] && nums[i] === nums[i + 2]) {
                found = true;
                result = false;
            }
        }
    }

    return result;

};

console.log(noTriples([1, 1, 2, 2, 1]));
console.log(noTriples([1, 1, 2, 2, 2, 1]));
console.log(noTriples([1, 1, 1, 2, 2, 2, 1]));
